pdfgen.c

```diff
@@ -2509,6 +2537,8 @@ int pdf_add_rgb24(struct pdf_doc *pdf, struct pdf_object *page, float x,
     if (!obj)
         return pdf->errval;
 
+    get_img_display_dimensions(width, height, &display_width,
```

Comment:
As you noted, I think we should change the prototype for the `pdf_add_rgb24` function to use `uint32_t width, uint32_t height`. On most platforms that is basically identical anyway, and it's more consistent.